xbmc/guilib/GUIWindow.cpp

```diff
@@ -141,16 +142,28 @@ bool CGUIWindow::LoadXML(const std::string &strPath, const std::string &strLower
     StringUtils::ToLower(strPathLower);
     if (!xmlDoc.LoadFile(strPath) && !xmlDoc.LoadFile(strPathLower) && !xmlDoc.LoadFile(strLowerPath))
     {
-      CLog::Log(LOGERROR, "unable to load:%s, Line %d\n%s", strPath.c_str(), xmlDoc.ErrorRow(), xmlDoc.ErrorDesc());
+      CLog::Log(LOGERROR, "Unable to load window XML: %s. Line %d\n%s", strPath.c_str(), xmlDoc.ErrorRow(), xmlDoc.ErrorDesc());
       SetID(WINDOW_INVALID);
       return false;
     }
-    m_windowXMLRootElement = (TiXmlElement*)xmlDoc.RootElement()->Clone();
+    m_windowXMLRootElement = static_cast<TiXmlElement*>(xmlDoc.RootElement()->Clone());
   }
   else
     CLog::Log(LOGDEBUG, "Using already stored xml root node for %s", strPath.c_str());
 
-  return Load(m_windowXMLRootElement);
+  // we must create a copy of the root element as we will manipulate it
+  // when loading and we don't want the original root element to change
+  TiXmlElement* pRootElementCopy = static_cast<TiXmlElement*>(m_windowXMLRootElement->Clone());
```

Comment:
I think it would be better to move the preparation of XML into a separate method `PrepareXML`. The result will be passed into `Load` which do what you expect, only read from XML. That way it should be clear what happens.